Update hardcoded kafka release name

genti-t opened the pull request:
This replaces the hardcoded value with user-specified ones, allowing to have multiple Kafka deployments.

alexvicegrab commented:
Great, thank you for this. We will need to update the unit tests to account for the new Kafka naming, so that they can support an arbitrary Kafka HLF name.

alexvicegrab commented:
Currently Travis CI is not fully working, not quite sure why. After updating your branch, try to run these 2 commands locally for unit and integration tests (latter ones should run on Minikube):

PYTHONPATH=. pytest --ignore=./integration --cov=. --cov-report xml:coverage.xml --cov-report term-missing

PYTHONPATH=. pytest -x -s ./integration

genti-t commented:
Unit test if fine now.
The integration test is failing on 
```
>           channels = (res.split("Channels peers has joined: ")[1]).split()
E           AttributeError: 'NoneType' object has no attribute 'split'

nephos/fabric/peer.py:221: AttributeError
```

genti-t commented:
I also checked the master branch  with:
```
PYTHONPATH=. pytest -x -s ./integration
```

```
2019-03-18 16:18:02.019 UTC [grpc] DialContext -> DEBU 036 scheme "" not registered, fallback to default scheme
2019-03-18 16:18:02.019 UTC [grpc] watcher -> DEBU 037 ccResolverWrapper: sending new addresses to cc: [{peer1-hlf-peer:7051 0  <nil>}]
2019-03-18 16:18:02.019 UTC [grpc] switchBalancer -> DEBU 038 ClientConn switching balancer to "pick_first"
2019-03-18 16:18:02.019 UTC [grpc] HandleSubConnStateChange -> DEBU 039 pickfirstBalancer: HandleSubConnStateChange: 0xc42046e2a0, CONNECTING
Error: error getting endorser client for channel: endorser client failed to connect to peer1-hlf-peer:7051: failed to create new connection: context deadline exceeded
command terminated with exit code 1

F

=================================================================================================================================== FAILURES ===================================================================================================================================
___________________________________________________________________________________________________________________ TestIntegrationDev.test_integration_dev ____________________________________________________________________________________________________________________

self = <test_dev.TestIntegrationDev object at 0x10ef5be80>

    def test_integration_dev(self):
        # Get options
        opts = load_config(self.CONFIG)

        # TODO: There should be a more elegant way of obtaining all the releases
        releases = (
            [key for key in opts["cas"].keys()]
            + [key + "-pg" for key in opts["cas"].keys()]
            + opts["orderers"]["names"]
            + [("cdb-" + key) for key in opts["peers"]["names"]]
            + [key for key in opts["peers"]["names"]]
        )

        # Run Fabric script
        check_cluster(
            self.CONTEXT
        )  # Dangerous operation, recheck we have not shifted context
>       runner_fabric(opts)

integration/test_dev.py:32:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
nephos/runners.py:109: in runner_fabric
    runner_peer(opts, upgrade=upgrade, verbose=verbose)
nephos/runners.py:132: in runner_peer
    create_channel(opts, verbose=verbose)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

opts = OrderedDict([('core', OrderedDict([('cluster', 'minikube'), ('chart_repo', 'stable'), ('dir_config', '/Users/genti/rep...', ['peer1']), ('channel_name', 'mychannel'), ('channel_profile', 'MyChannel'), ('secret_channel', 'hlf--channel')]))]), verbose = False

    def create_channel(opts, verbose=False):
        """Create Channel for Peer.

        Args:
            opts (dict): Nephos options dict.
            verbose (bool): Verbosity. False by default.
        """
        peer_namespace = get_namespace(opts, opts["peers"]["msp"])
        ord_namespace = get_namespace(opts, opts["orderers"]["msp"])
        channel = opts["peers"]["channel_name"]
        # Get orderer TLS status
        ord_name = random.choice(opts["orderers"]["names"])
        # TODO: This should be a function
        cmd_suffix = peer_channel_suffix(opts, ord_name, verbose=verbose)

        for index, release in enumerate(opts["peers"]["names"]):
            # Get peer pod
            pod_ex = get_pod(peer_namespace, release, "hlf-peer", verbose=verbose)

            # Check if the file exists
            has_channel = False
            while not has_channel:
                has_channel = get_channel_block(
                    pod_ex, ord_name, ord_namespace, channel, cmd_suffix
                )
                if not has_channel:
                    pod_ex.execute(
                        (
                            "bash -c 'peer channel create "
                            + "-o {orderer}-hlf-ord.{ns}.svc.cluster.local:7050 "
                            + "-c {channel} -f /hl_config/channel/{channel}.tx {cmd_suffix}'"
                        ).format(
                            orderer=ord_name,
                            ns=ord_namespace,
                            channel=opts["peers"]["channel_name"],
                            cmd_suffix=cmd_suffix,
                        )
                    )
            res, _ = pod_ex.execute("peer channel list")
>           channels = (res.split("Channels peers has joined: ")[1]).split()
E           AttributeError: 'NoneType' object has no attribute 'split'

nephos/fabric/peer.py:221: AttributeError
```

Am I missing something?

alexvicegrab commented on tests/fabric/test_ord.py:
```diff
@@ -103,7 +103,7 @@ def test_ord(self, mock_check_ord, mock_helm_install, mock_helm_upgrade):
     def test_ord_kafka(self, mock_check_ord, mock_helm_install, mock_helm_upgrade):
         OPTS = deepcopy(self.OPTS)
-        OPTS["orderers"]["kafka"] = {"pod_num": 42}
+        OPTS["orderers"]["kafka"] = {"name": "kafka-hlf", "pod_num": 42}
```
Great, thanks!

alexvicegrab commented:
Aha, @genti-t, not sure, something about the connection is wrong, namely

"""
Error: error getting endorser client for channel: **endorser client failed to connect to peer1-hlf-peer:7051**: failed to create new connection: context deadline exceeded
"""

alexvicegrab commented:
Which version of Minikube are you using? I had issues with older versions of Minikube and needed to delete my minikube and recreate with latest before I managed to get it working. Something about KubeDNS that fails unfortunately.

alexvicegrab commented:
This is indeed why I had the Travis CI checks, once these are sorted and work here, it will be much easier to do the tests automatically.

alexvicegrab commented:
Alternatively, you should be able to activate TravisCI on your own Fork by adding TravisCI via your Marketplace and using that to run an analysis on your own branch, which will be a good first sanity check everything works.
